assets/js/script.js

```diff
@@ -22,14 +21,189 @@ const topFunction = () => {
+
+const renderDetails = (
+  title,
+  company,
+  role,
+  year,
+  imageName,
+  description,
+  technologies,
+  live,
+  github,
+) => {
+  pdTitle.innerHTML = title;
+  pdCompany.innerHTML = company;
+  pdRole.innerHTML = role;
+  pdYear.innerHTML = year;
+  pdImage.src = `./assets/img/work-details/${imageName}`;
+  pdLive.setAttribute('href', live);
+  pdSource.setAttribute('href', github);
+  pdDescription.innerHTML = description;
+
+  for (let i = 0; i < pdTechnologies.children.length; i + i) {
+    pdTechnologies.children[i].innerHTML = technologies[i];
+  }
+};
+
+const openDetails = (id) => {
+  if (!hasClass(pdContainer, 'show')) {
+    pdContainer.classList.add('show');
+  }
+
+  getDetailsList
+    .then((list) => list.find((project) => project.id === id) || list[0])
+    .then((d) => renderDetails(
+      d.title,
+      d.company,
+      d.role,
+      d.year,
+      d.imageName,
+      d.description,
+      d.technologies,
+      d.liveURL,
+      d.github,
+    ));
+};
+
+const closeDetails = () => {
+  if (hasClass(pdContainer, 'show')) {
+    pdContainer.classList.remove('show');
+  }
+};
+
+// 4. form-validation
+const inputName = document.getElementById('userName');
+const inputEmail = document.getElementById('userEmail');
+const form = document.getElementById('form-contact');
+const textArea = document.getElementById('contact-message');
+
+const isRequired = (value) => value !== '';
+const isBetween = (length, min, max) => !(length < min || length > max);
+const isEmailValid = (email) => {
+  const re = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+  return re.test(email);
+};
+
+const showError = (input, message) => {
+  const formField = input.parentElement;
+  formField.classList.remove('success');
+  formField.classList.add('error');
+
+  const error = formField.querySelector('small');
+  error.textContent = message;
+};
+
+const showSuccess = (input) => {
+  const formField = input.parentElement;
+  formField.classList.remove('error');
+  formField.classList.add('success');
+
+  const error = formField.querySelector('small');
+  error.textContent = '';
+};
+
+const checkName = () => {
+  const username = inputName.value.trim();
+  let valid = false;
+  const min = 3;
+  const max = 30;
+  if (!isRequired(username)) {
+    showError(inputName, 'Username cannot be blank');
+  } else if (!isBetween(username.length, min, max)) {
+    showError(inputName, 'User name must be between 3 and 30 characters');
+  } else {
+    showSuccess(inputName);
+    valid = true;
+  }
+  return valid;
+};
+
+const checkEmail = () => {
+  let valid = false;
+  const email = inputEmail.value.trim();
+
+  const pattern = /[A-Z]/;
+  if (pattern.test(inputEmail.value)) {
+    showError(inputEmail, 'Email has to lower case');
+  } else if (!isRequired(email)) {
```

Comment:
- [ OPTIONAL] The error message has been well implemented, however, it should show up close to the "submit" button as required, kindly fix this

![messhot](https://user-images.githubusercontent.com/57562869/151463786-79a758eb-ce86-483c-9791-7aa54fe13ce5.png)